optuna/storages/_rdb/storage.py

```diff
@@ -866,6 +882,27 @@ def _build_frozen_trial_from_trial_model(self, trial: "models.TrialModel") -> Fr
             values = None
 
         params = sorted(trial.params, key=lambda p: p.param_id)
+        unsorted_intermediates: Dict[int, Dict[int, float]] = defaultdict(dict)
+        for v in trial.intermediate_values:
+            unsorted_intermediates[v.step][
+                v.intermediate_value_index
+            ] = v.stored_repr_to_intermediate_value(
+                v.intermediate_value, v.intermediate_value_type
+            )
+        intermediate_values: Dict[int, Union[float, Sequence[float]]] = {}
+        if unsorted_intermediates:
+            num_intermediate_values_at_same_step = len(unsorted_intermediates[v.step])
+            if num_intermediate_values_at_same_step != 1:
+                for step, intermediate_indices_and_values in unsorted_intermediates.items():
+                    intermediate_values[step] = tuple(
+                        intermediate_value
+                        for _, intermediate_value in sorted(
+                            intermediate_indices_and_values.items()
+                        )
+                    )
+            else:
+                for step, intermediate_indices_and_values in unsorted_intermediates.items():
+                    intermediate_values[step] = intermediate_indices_and_values[0]
 
```

Comment:
How about adding tests cases for these logics?

Reply:
Thank you for your comment. I added test cases for these logics in tests/storages_tests/rdb_tests/test_storage.py.